class-unlist-posts-admin.php

```diff
@@ -142,6 +145,75 @@ public function save_meta( $post_id ) {
 		update_option( 'unlist_posts', $hidden_posts );
 	}
 
+	/**
+	 * Add 'Unlisted' post status to post list items.
+	 *
+	 * @param Array $states   An array of post display states.
+	 * @param Post  $post     The current post object.
+	 *
+	 * @return Array  $states An updated array of post display states.
+	 */
+	function add_unlisted_post_status( $states, $post ) {
+		// Bail if the unlisted post filter is active, to avoid redundancy.
+		if ( is_admin() && 'unlisted' == $_GET['post_status'] ) {
+			return;
+		}
+
+		// Get the list of unlisted post IDs from the options table.
+		$unlisted_posts = maybe_unserialize( get_option( 'unlist_posts' ) );
+
+		// Check if this post is unlisted and mark it as so if appropriate.
+		if ( in_array( $post->ID, $unlisted_posts ) ) {
+			$states[] = __( 'Unlisted', 'unlist-posts' );
+		}
+
+		return $states;
+	}
+
+	/**
+	 * Add 'Unlisted' filter to the post list.
+	 *
+	 * @param Array $views   An array of post list filters.
+	 *
+	 * @return Array $views  An updated array of post list filters.
+	 */
+	function add_unlisted_post_filter( $views ) {
+		// Get the list of unlisted post IDs from the options table.
+		$unlisted_posts = maybe_unserialize( get_option( 'unlist_posts' ) );
+		$unlisted_count = count( $unlisted_posts );
+
+		// Mark 'Unlisted' filter as the current filter if it is.
+		$link_attributes = '';
+		if ( is_admin() && 'unlisted' == $_GET['post_status'] ) {
+			$link_attributes = 'class="current" aria-current="page"';
+		}
+
+		$views['unlisted'] = '<a href="edit.php?post_status=unlisted&post_type=post" ' . $link_attributes . '>' . __( 'Unlisted', 'unlist-posts' ) . ' <span class="count">(' . $unlisted_count . ')</span></a>';
```

Comment:
`$unlisted_count` should be escaped using `esc_html`

Reply:
Fixed!